### ISSUE TITLE
Segfault on startup

### ISSUE THREAD

**Issue by givepiesachance** [atom/atom#18103]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: https://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

When I run atom (including atom --safe) it crashes and outputs:

/usr/bin/atom: line 157: 11692 Segmentation fault      (core dumped) nohup "$ATOM_PATH" --executed-from="$(pwd)" --pid=$$ "$@" > "$ATOM_HOME/nohup.out" 2>&1

~/.atom/nohup.out is present but empty.

### Versions

Tested on stable and beta:

apm-beta --version 
apm  2.1.0
npm  6.2.0
node 8.9.3 x64
atom 1.31.0-beta0
python 2.7.15+
git 2.17.1

apm --version
apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.30.0
python 2.7.15+
git 2.17.1

**Comment by abrumloop** [comment]
same problem for me (downloaded the rpm package on F29). I assumed this was caused by Fedora Beta.

apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.30.0
python 2.7.15
git 2.19.0

nohup.out is empty except 11 empty lines.

**Comment by steveward** [comment]
Thanks for the report @givepiesachance. This issue looks similar to https://github.com/atom/atom/issues/16784. 

Have you tried [clearing the saved state](https://flight-manual.atom.io/hacking-atom/sections/debugging/#clearing-saved-state) to see if that fixes this crash?

**Comment by givepiesachance** [comment]
$ atom --clear-window-state
$ /usr/bin/atom: line 157:  2071 Segmentation fault      (core dumped) nohup "$ATOM_PATH" --executed-from="$(pwd)" --pid=$$ "$@" > "$ATOM_HOME/nohup.out" 2>&1

**Comment by rsese** [comment]
What operating system are you running @givepiesachance?

**Comment by pgbross** [comment]
`atom 1.31.0-beta0 (electron 2.0.7)`

I get the same problem after upgrading the OS to Fedora 29 Beta with the segmentation fault and notably it occurs both in atom and vscode. 

However a newer "insiders" build of vscode that uses Electron 2.0.9 does not fail with a segmentation fault. Perhaps an update to Atom to use the more recent version of Electron will help.

--philip

**Comment by pgbross** [comment]
I note there was a commit `Updated Electron fixes segfault on glibc 2.28` `b34cefa` on Sep 10th, so will try following the [linux install](https://flight-manual.atom.io/hacking-atom/sections/hacking-on-atom-core/#platform-linux) documentation.

Ok there is no segmentation fault if I compile and install 1.32.0-beta0 locally.

Here are the versions I used:

```
$ apm-beta --version
apm  2.1.2
npm  6.2.0
node 8.9.3 x64
atom 1.32.0-beta0
python 2.7.15
git 2.19.0
```
```
$ atom-beta --version
Atom    : 1.32.0-beta0
Electron: 2.0.9
Chrome  : 61.0.3163.100
Node    : 8.9.3
```
```
$ uname -a
Linux ab350n.localdomain 4.19.0-0.rc5.git1.2.fc30.x86_64 #1 SMP Tue Sep 25 15:35:01 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by michel-slm** [comment]
Confirmed that 1.32 stable carries the fix as well
